app/src/components/map/InteractionLayer.jsx

```diff
@@ -199,6 +212,48 @@ const InteractionLayer = ({ baseStyles, interactionStyles }) => {
           .setContent("<div style='color: red;'>Failed to fetch GeoJSON</div>")
           .openOn(mapRef.current);
       }
+
+      // timing on highlighting upstream and downstream features is a little
+      // tricky because the fetch is a little slow, giving things time to get out
+      // of sync. 
+      // If we clear all the existing upstream and downstream features
+      // before fetching the new ones, there are a few seconds during the fetch
+      // where a user can click a second time and cause two seperate upstream
+      // and downstream highlghts to appear, due to running the clear only before
+      // the fetch.
+      // On the other hand, if we fetch, clear, and display one set of features before
+      // fetching, clearing, and displaying the other set, the second clear may accidentally
+      // remove some of the just-fetched highlights.
+      // so both sets of features have to be fetched, then cleared, then displayed in tandem.
+      try {
+        // fetch upstream and downstream features
+        const downstreamList = await fetchDownstreams(properties.subid);
+        const upstreamList = await fetchUpstreams(properties.subid);
```

Comment:
If you wanted to speed this up a bit you can kick off both queries and await both. Something like:

```js
const downStreamList, upstreamList = await Promise.all([fetchDownstreams(properties.subid), fetchUpstreams(properties.subid)]);
```